packages/mui-base/src/useNumberField/useNumberField.ts

```diff
@@ -40,7 +40,7 @@ import { mergeReactProps } from '../utils/mergeReactProps';
  *
  * - [useNumberField API](https://mui.com/base-ui/react-number-field/hooks-api/#use-number-field)
  */
-export function useNumberField(params: NumberFieldProps): UseNumberFieldReturnValue {
+export function useNumberField(params: NumberFieldRootProps): UseNumberFieldReturnValue {
```

Comment:
This is out of scope of this PR, but I just noticed: hooks need parameters called `<hookName>Parameters` to be found by the API docs generator.
Also, since hooks are lower-level primitives, we tend not to import types from components in hooks (only the other way around). It's just a convention that helps to avoid dependency cycles.